feat(curriculum): add Artifact Provenance Auditor workshop (#68860)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-artifact-provenance-auditor/6a57c331ae94fa4631b60873.md

---
id: 6a57c331ae94fa4631b60873
title: Step 20
challengeType: 1
dashedName: step-20
---

# --description--

> Generate a summary for the artifact with an `id` of `101` and log it.

Reviewer: ```suggestion
Generate a summary for artifact `101` and log the result.
```

> You have completed the Artifact Provenance Auditor workshop!

Reviewer: ```suggestion
Congratulations! You have successfully completed the Artifact Provenance Auditor workshop.
```

---
id: 6a57c331ae94fa4631b60873
title: Step 20
challengeType: 1
dashedName: step-20
---

# --description--

Generate a summary for the artifact with an `id` of `101` and log it.

You have completed the Artifact Provenance Auditor workshop!

# --before-each--

```js
const spy = __helpers.spyOn(console, "log");
const getLogs = () => spy.calls.map((call) => call?.[0]);
```

# --hints--

You should call `buildSummary` with `101`.

```js
assert.match(__helpers.removeJSComments(code), /buildSummary\s*\(\s*101\s*\)/);
```

You should log the result of `buildSummary(101)`.

```js

> assert.include(getLogs(), buildSummary(101));

Reviewer: Replace this with regex, as explained in earlier comment.
